docs/web-apps/automated-testing/pre-run-executables.md

```diff
@@ -0,0 +1,308 @@
+
+```
+desired_capabilities['prerun'] = {
+    'executable':'http://myscriptstorage.com/disable_fraud.sh',
+    'background': 'false'
+}
+```
+
+This example accesses the same script from Sauce Storage:
+
+```
+desired_capabilities['prerun'] = {
+    'executable':'sauce-storage:disable_fraud.sh',
+    'background': 'false'
+}
+```
+
+## Setting Up Silent Mode
+One of the arguments that can be added to the `prerun` capability in your test configuration options is `--silent`, which can be abbreviated to `/S`. This argument will "silently" install the executable without any modal dialogs or other installation processes showing up in your tests. Here's an example of how you would set the prerun capability to run in silent mode:
+
+```
+"prerun": {
+    "executable": "http://url.to/your/executable.exe",
+    "args": [ "/S", "-a", "-q" ],
+    "background": true,
+    "timeout": 120
+}
+```
+
+## Setting Basic Authentication in Safari
+In Safari, when you try to navigate to a URL that uses basic HTTP authentication, a warning dialog pops up that will derail your automated tests unless you can resolve it. You can use a short shell script that you execute as a pre-run executable to configure Safari to not show this message.
+
+>**NOTE:** You can only use this for Safari versions up to and including 10.x.
+
+This script clears the Warn when visiting a fraudulent website option in Safari's preferences, so that your automated tests that use basic authentication over HTTP will execute without triggering the warning.
+
+```
+#!/bin/bash
+defaults write com.apple.Safari WarnAboutFraudulentWebsites false
+```
+
+[This script](https://gist.githubusercontent.com/saucyallison/3a73a4e0736e556c990d/raw/d26b0195d48b404628fc12342cb97f1fc5ff58ec/disable_fraud.sh) is hosted as a Gist on GitHub.
+
+### Setting the `prerun` Capability
+You can host this script in your own repository, or in temporary Sauce Storage, and refer to it within the prerun capability of your tests as described in the [Test Configuration Options](https://wiki.saucelabs.com/display/DOCSDEV/Test+Configuration+Options#TestConfigurationOptions-Pre-runExecutables) topic. This Python example refers to the GitHub Gist location:
+
+```
+desired_capabilities['prerun'] = {
+  'executable':'https://gist.githubusercontent.com/saucyallison/3a73a4e0736e556c990d/raw/d26b0195d48b404628fc12342cb97f1fc5ff58ec/disable_fraud.sh',
+    'background': 'false'
+}
+```
+
+## Downloading Files to a VM Prior to Testing
+
+You can use a pre-run executable script to download files from a public location to the Sauce Labs virtual machine running your tests. This topic contains example scripts for downloading remote files on different operating systems, and details configuring the prerun capability in your tests.
+
+<Tabs
+  defaultValue="macOS"
+  values={[
+    {label: 'macOS', value: 'macOS'},
+    {label: 'Windows', value: 'Windows'},
+    {label: 'Windows XP', value: 'Windows XP'},
+    {label: 'Linux', value: 'Linux'},
```

Comment:
Are these working in your local? I thought that the values had to be all lowercase and a non-breaking string?

Reply:
Yep, they work for me